### PR TITLE
Hydrate errors on fieldsets and base field

### PR DESCRIPTION
**Pull request by joeosburn**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -101,6 +101,11 @@ const hydrate = (fields, doc, type) => {
     }
   }
 
+  const root = dotProp.get(doc, '');
```
Comment (shanebo) [inline lib/index.js]: @joeosburn what's the purpose of this line?

Comment (joeosburn) [reply lib/index.js]: What are you getting at exactly? I'm just trying to not call `dotProp.get(doc, '');` twice

Comment (shanebo) [reply lib/index.js]: We can discuss tomorrow.

File: lib/index.js [hunk 2]
```diff
@@ -101,18 +101,17 @@ const hydrate = (fields, doc, type) => {
...
-  Object
+  const keyMap = Object
     .keys(dottedFields)
     .filter(key => key.includes('_attributes.name'))
     .map(key => ({
       key: dottedFields[key],
       multiple: dottedFields[key].match(/\.(\d+)\./)
     }))
+
+  keyMap.push({key: '', multiple: false});
```
Comment (joeosburn) [inline lib/index.js]: @shanebo Here is an alternative version

**Review by shanebo — APPROVED** [review]

**Pull request merged**